### ISSUE TITLE
Sign of angle and resulting [Counter]clockwise rotations of `SteerableBasis2D` coefficients

### ISSUE THREAD

**Issue by chris-langfield** [ComputationalCryoEM/ASPIRE-Python#747]
Arising from discussion in #736, which introduced the explicit calculation of rotated *real* basis coefficients, we found a possible inconsistency between theory and implementation:

Achieving a **CCW** rotation of the resulting image requires multiplication by `exp(i k theta)` in practice, while the exponent is generally negated in the papers.

Similarly, the implemented real formula achieving the same result appears to also have the sign flipped relative to what would be derived from the `exp(-i k theta)` formula.

If you set `theta = -theta` in the derivation in the post below this, I believe you will arrive at the formula now used in our code because of the behavior of `sin` as an odd function. 

TLDR: It seems that `exp(i k theta)` (and the corresponding derived real formula) is the one that actually works to achieve a CCW rotation. Does this mean that:
- Our convention should actually be CW (meaning the theory is correct and we should flip the signs of the implemented code)
- The theory should actually have `exp(i k theta)` and not `exp(-i k theta)`
- There is a bug in our basis coefficients themselves such that they are rotating the wrong direction, but `exp(- i k theta)` is correct, and we had to flip the sign to account for this possible bug

**Comment by chris-langfield** [comment]
[RealvsComplexCoeffsASPIRE.pdf](https://github.com/ComputationalCryoEM/ASPIRE-Python/files/9992617/RealvsComplexCoeffsASPIRE.pdf)

![Screen Shot 2022-11-11 at 1 34 11 PM](https://user-images.githubusercontent.com/34426450/201408150-56c8b81e-484a-4759-90cb-bbc606c7e17b.png)

**Comment by garrettwrong** [comment]
I think flipping the of `sgns` array ranges in the basis indices will reverse the effect of rotation. Perhaps the indices mappings or loop(s) are backwards somewhere.

**Comment by chris-langfield** [comment]
The MATLAB version below matches the math above (so it's flipped from our Python code)

https://github.com/PrincetonUniversity/aspire/blob/72ec72eb2608c8308592a49f8f4f6aac986bd938/projections/fb_rotate.m#L29-L42

**Comment by garrettwrong** [comment]
The convention doc and docstring in matlab doesn't mention which way the rotation occurs...

Were we in agreement that formula is clockwise...?

**Comment by chris-langfield** [comment]
Displaying the image in MATLAB shows a CCW rotation

**Comment by garrettwrong** [comment]
Hrrm, maybe this a plotting imshow origin thing?

**Comment by garrettwrong** [comment]
I think we should just look at a small one (pixel) hot array, evaluate_t, rotate, evaluate, and look at the output array.  Let's start with Python.

**Comment by garrettwrong** [comment]
I'm fine with saying that (some of) the papers are wrong if you are.

https://mathworld.wolfram.com/ComplexRotation.html#:~:text=A%20complex%20rotation%20is%20a,of%20points%20the%20complex%20plane.

⛽ 😇

**Comment by chris-langfield** [comment]
Hmm... but the equation in MATLAB is *real*

**Comment by garrettwrong** [comment]
The equation in MATLAB doesn't specify which way it is rotating...

I feel like we have it on pretty good authority that if we take a thing and apply the exponential formula `e^{i theta}` for positive theta, it rotates the thing CCW.  Do you disagree with this or not?

**Comment by chris-langfield** [comment]
I think that page is referring to the position of a complex number *in the complex plane*. I'm not ready to jump from there to saying that our real 2D functions are rotated CCW if their complex coeffs are multiplied by that factor

**Comment by garrettwrong** [comment]
Okay, that's fair I guess.

_We are dealing with coefficients, not functions_.  Have you considered that the functions might rotate opposite the angle intended to rotate the image (coefs)... 

<img width="602" alt="Screen Shot 2022-11-22 at 12 57 26 PM" src="https://user-images.githubusercontent.com/47759732/203387071-caa1fcfe-a85b-4b86-8297-658605910f06.png">

I thought I found a piece of damning evidence, but alas, no ccw vs cw was mentioned  😂 .

<img width="534" alt="Screen Shot 2022-11-22 at 12 51 54 PM" src="https://user-images.githubusercontent.com/47759732/203386009-cb0f2f95-b48b-4446-8a91-5dca13524ba2.png">

**Comment by garrettwrong** [comment]
Scanning issues... I think we can close this.  The only open item was narrowing down why some MATLAB plots were rotated with the (CCW) convention we are using in Python. Considering we are achieving the same result, I'm not sure going into the legacy code just for this is going to gain us anything.  If it comes up again we can just reopen.
